webapp/converter.xqm

```diff
@@ -1,5 +1,19 @@
 module namespace e = 'some.namespace';
 
+declare
+  %rest:path("/v1tov2")
+  %rest:POST("{$xmlString}")
+  %rest:consumes('text/plain')
+  %output:method("xml")
+  %output:indent("no")
+  %output:omit-xml-declaration("no")
+  %output:doctype-public("-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD with MathML3 v1.2 20190208//EN")
+  %output:doctype-system("JATS-archivearticle1-mathml3.dtd")
+function e:v1tov2string($xmlString as xs:string) {
+  let $xml := fn:parse-xml($xmlString) 
```

Comment:
> let $xml := fn:parse-xml($xmlString) 

Not sure if this is necessary. You _can_ pass the first parameter of [`xslt:transform`](https://docs.basex.org/wiki/XSLT_Module#xslt:transform) as a string instead of an XML document if you want to.  Although it might be useful like it is depending on how you want to catch an error.

XQuery has try catch which might be of use, depending on the ideal response when there is an error. Example:
```
try { (: do something here :) }
catch * {'Some text: ' || $err:code || ' Some more text: ' || $err:description}
```

[`xslt:transform-text`](xslt:transform-text) might also be of interest if you want this passed back as a string. 

If that's desirable you could use that function instead and remove all the `%output:method`s. The xml declaration and doctype declaration could then just be prefixed to the string that's produced (could use `concat()` or `'text' || $xml` syntax to do so).

Reply:
All great feedback, thanks Fred. I'll look into this now.